packages/flutter/lib/src/material/filled_button.dart

```diff
@@ -0,0 +1,753 @@
+  /// * `minimumSize` - Size(64, 40)
+  /// * `fixedSize` - null
+  /// * `maximumSize` - Size.infinite
+  /// * `side` - null
+  /// * `shape` - StadiumBorder()
+  /// * `mouseCursor`
+  ///   * disabled - SystemMouseCursors.basic
+  ///   * others - SystemMouseCursors.click
+  /// * `visualDensity` - Theme.visualDensity
+  /// * `tapTargetSize` - Theme.materialTapTargetSize
+  /// * `animationDuration` - kThemeChangeDuration
+  /// * `enableFeedback` - true
+  /// * `alignment` - Alignment.center
+  /// * `splashFactory` - Theme.splashFactory
+  ///
+  /// The default padding values for the [FilledButton.icon] factory are slightly different:
+  ///
+  /// * `padding`
+  ///   * `textScaleFactor <= 1` - start(12) end(16)
+  ///   * `1 < textScaleFactor <= 2` - lerp(start(12) end(16), horizontal(8))
+  ///   * `2 < textScaleFactor <= 3` - lerp(horizontal(8), horizontal(4))
+  ///   * `3 < textScaleFactor` - horizontal(4)
+  ///
+  /// The default value for `side`, which defines the appearance of the button's
+  /// outline, is null. That means that the outline is defined by the button
+  /// shape's [OutlinedBorder.side]. Typically the default value of an
+  /// [OutlinedBorder]'s side is [BorderSide.none], so an outline is not drawn.
+  ///
+  @override
+  ButtonStyle defaultStyleOf(BuildContext context) {
+    switch (_variant) {
+      case _FilledButtonVariant.filled:
+        return _FilledButtonDefaultsM3(context);
+      case _FilledButtonVariant.tonal:
+        return _FilledTonalButtonDefaultsM3(context);
+    }
+  }
+
+  /// Returns the [FilledButtonThemeData.style] of the closest
+  /// [FilledButtonTheme] ancestor.
+  @override
+  ButtonStyle? themeStyleOf(BuildContext context) {
+    return FilledButtonTheme.of(context).style;
+  }
+}
+
+EdgeInsetsGeometry _scaledPadding(BuildContext context) {
+  return ButtonStyleButton.scaledPadding(
+    const EdgeInsets.symmetric(horizontal: 16),
+    const EdgeInsets.symmetric(horizontal: 8),
+    const EdgeInsets.symmetric(horizontal: 4),
+    MediaQuery.maybeOf(context)?.textScaleFactor ?? 1,
+  );
+}
+
+@immutable
+class _FilledButtonDefaultColor extends MaterialStateProperty<Color?> with Diagnosticable {
+  _FilledButtonDefaultColor(this.color, this.disabled);
+
+  final Color? color;
+  final Color? disabled;
+
+  @override
+  Color? resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.disabled)) {
+      return disabled;
+    }
+    return color;
+  }
+}
+
+@immutable
+class _FilledButtonDefaultOverlay extends MaterialStateProperty<Color?> with Diagnosticable {
+  _FilledButtonDefaultOverlay(this.overlay);
+
+  final Color overlay;
+
+  @override
+  Color? resolve(Set<MaterialState> states) {
+    if (states.contains(MaterialState.hovered)) {
+      return overlay.withOpacity(0.08);
+    }
+    if (states.contains(MaterialState.focused) || states.contains(MaterialState.pressed)) {
+      return overlay.withOpacity(0.24);
```

Comment:
Fyi, as per material 3 spec, opacity for this state is at 12%. This applies to other buttons too!
![Reference](https://lh3.googleusercontent.com/uEnPoGsoGM2IQQY-78Grz3A6lD0k-JVkEXLk-14dhf0M0pCRKhkhMuoOtqbVfe7I4KGRxEWVZ2UvlIw0jLTABvmWPOdzqxBKm11LGJwtRl7hQA=s0)
```suggestion
      return overlay.withOpacity(0.12);
```